src/pointarray.cpp

```diff
@@ -387,10 +506,53 @@ V3d PointArray::pickVertex(const V3d& cameraPos,
                            double longitudinalScale, double* distance,
                            std::string* info) const
 {
+    // NOTE rayOrigin here is not cameraPos but something else
+    // This routine uses cameraPos as the rayOrigin point
```

Comment:
(The idea was, if you can guess roughly where the user is looking, you should probably select something at a similar depth into the scene.)